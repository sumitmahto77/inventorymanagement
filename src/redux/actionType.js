// // export const GET_USERS ="GET_USERS";
// // export const ADD_USER = "ADD_USER";

// export const LOGIN_START = "LOGIN_START";
// export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
// export const LOGIN_FAIL = "LOGIN_FAIL";

// export const LOGOUT_USER = "LOGOUT_USER";

// export const REGISTER_START = "REGISTER_START";
// export const REGISTER_SUCCESS = "REGISTER_SUCCESS";
// export const REGISTER_FAIL = "LOGIN_FAIL";



