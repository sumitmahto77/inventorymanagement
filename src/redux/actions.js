import * as types from './actionType';
import axios from 'axios';

// const getUsers = (users) => ({
//     type: types.GET_USERS,
//     payload: users,
// });

// const userAdded = () =>({
//     type:types.ADD_USER,

// });

// export const loadUsers = () => {
//     return function(dispatch){
//         axios.get("http://localhost:4000/users")
//         .then((response) => {
//             console.log("response", response);
//             dispatch(getUsers(response.data));
//         })
//         .catch((error) => console.log(error));
//     };
// };

// export const addUser = (user) => {
//     return function (dispatch){
//         axios
//             .post("http://localhost:4000/users",user)
//             .then((response) =>{
//                 console.log("response",response);
//                 dispatch(userAdded());
//             })
//             .catch((error) => console.log(error));
//     };
// };


const loginStart = () => ({
    type : types.LOGIN_START,
});

const loginSuccess = (token) => ({
    type : types.LOGIN_SUCCESS,
    payload : token,
});

const loginFail = (error) => ({
    type : types.LOGIN_FAIL,
    payload : error,
});

const registerStart = () => ({
    type : types.REGISTER_START,
});

const registerSuccess = (token) => ({
    type : types.REGISTER_SUCCESS,
    payload : token,
});

const registerFail = (error) => ({
    type : types.REGISTER_FAIL,
    payload : error,
});

export const logout = () =>({
    type : types.LOGOUT_USER,

})


export const loginInitiate = (email,password) => {
    return function(dispatch){
        dispatch(loginStart());
        axios
        .post("http://localhost:4000/login",{
            email,
            password,
        })
        .then((response) => {
            console.log("response", response);
            dispatch(loginSuccess(response.data.access_token))
            
        })
        .catch((error) => dispatch(loginFail(error.response.data.message)));
    };
};
export const registerInitiate = (user) => {
    return function(dispatch){
        dispatch(registerStart());
        axios
        .post("http://localhost:4000/users",user)
        .then((response) => {
            console.log("response", response);
            dispatch(registerSuccess(response.data))
            
        })
        .catch((error) => dispatch(registerFail(error.response.data.message)));
    };
};
