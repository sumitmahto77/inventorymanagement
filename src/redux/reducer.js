import * as types from './actionType';

const initialAuthState ={
    user : null,
    loading : false,
    error : null,
    token : null
};

const initialProdState = {
    products : [],
    loading : true,
};

export const authReducer = (state = initialAuthState, action) => {
    switch (action.type){
        case types.LOGIN_START:
        case types.REGISTER_START:
            return{
                ...state,
                loading : true,
            };
        case types.LOGIN_SUCCESS:
        case types.REGISTER_SUCCESS:
            return{
                ...state,
                loading: false,
                user :action.payload.user,
                token : action.payload.accessToken,
            };
        case types.LOGIN_FAIL:
        case types.REGISTER_FAIL:
            return{
                ...state,
                loading :false,
                error : action.payload,
            };

        case types.LOGOUT_USER:
            return{
                ...state,
                user : null,
                token :null,
            };

        default:
            return state;
    }
};

export const prodReducer =(state=initialProdState,action) => {
    switch (action.type){
        case types.GET_PRODUCTS:
            return {
                ...state,
                products : action.payload,
                loading: false,
            };
        case types.ADD_PRODUCT:
            return{
                ...state
            }
        case types.DELETE_PRODUCT:
            return{
                ...state,
                loading :false,
            }

        default:
            return state;
    }

}
