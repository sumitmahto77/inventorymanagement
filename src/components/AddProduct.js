import React from 'react';
import styled from 'styled-components';
import {mobile} from "../responsive";
import { useDispatch } from 'react-redux';
import { addProduct } from '../redux/actions';
import {useFormik} from 'formik';
import * as Yup from 'yup';
import PriorityHighIcon from '@mui/icons-material/PriorityHigh';
import { Prompt } from 'react-router';





const Container = styled.div`
  width: 100vw;
  height: 100vh;
  background: linear-gradient(
      rgba(255, 255, 255, 0.5),
      rgba(255, 255, 255, 0.5)
    ),
    url("https://images.pexels.com/photos/6984650/pexels-photo-6984650.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940")
      center;
  background-size: cover;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const Wrapper = styled.div`
  width: 40%;
  padding: 20px;
  background-color: white;
  ${mobile({ width: "75%" })}
`;

const Title = styled.h1`
  font-size: 24px;
  font-weight: 300;
`;

const Form = styled.form`
  display: flex;
  flex-direction: column;
`;

const Input = styled.input`
  flex: 1;
  min-width: 40%;
  margin: 10px 0;
  padding: 10px;
`;

const Button = styled.button`
  width: 40%;
  border: none;
  padding: 15px 20px;
  background-color: teal;
  color: white;
  cursor: pointer;
  margin-bottom: 10px;
  border-radius: 1vw;
`;

const validationSchema = Yup.object({
  name:Yup.string().required(<PriorityHighIcon/>),
  imgLink:Yup.string().required(<PriorityHighIcon/>),
  description:Yup.string().required(<PriorityHighIcon/>),
  manufacturer:Yup.string().required(<PriorityHighIcon/>),
  price:Yup.number().required(<PriorityHighIcon/>).integer().positive(),
  quantity:Yup.number().required(<PriorityHighIcon/>).integer().positive()
});

const AddProduct = () => {
  const dispatch = useDispatch();
  const {handleSubmit, handleChange,handleReset, values, errors,touched} = useFormik({
    initialValues:{
      name:"",
      imgLink:"",
      description:"",
      manufacturer:"",
      price:"",
      quantity:""
    },
    validationSchema,
    onSubmit(values){
      dispatch(addProduct({
        "name":values.name,
        "img":values.imgLink,
        "description" : values.description,
        "manufacturer" : values.manufacturer,
        "price" : values.price,
        "quantity" : values.quantity,
        "count" : 0
      }));
      alert("Product Added Successfully!");
      handleReset();

    }
  });

  return (
    <div>
        <Container>
          <Prompt when={!values.name || !values.imgLink || !values.description || !values.manufacturer || !values.price || !values.quantity} 
          message={"Do you want to discard the changes?"} />
          <Wrapper>
            <Title>Enter product details</Title>
            <Form onSubmit={handleSubmit}>
              {touched.name && errors.name ? <div style={{color:"red",display:'inline'}}>{errors.name}</div> :<></>}
              <Input type="text"  name="name" placeholder='name'  onChange={handleChange} value={values.name}/>
              
              {touched.imgLink && errors.imgLink ? <div style={{color:"red",display:'inline'}}>{errors.imgLink}</div> :<></>}
              <Input type="text" name="imgLink" placeholder='Image Link' onChange={handleChange} value={values.imgLink}/>
              
              {touched.description && errors.description ? <div style={{color:"red",display:'inline'}}>{errors.description}</div> :<></>}
              <Input type="text" name="description" placeholder='Description' onChange={handleChange} value={values.description}/>
              
              {touched.manufacturer && errors.manufacturer ? <div style={{color:"red",display:'inline'}}>{errors.manufacturer}</div> :<></>}
              <Input type="text" name="manufacturer" placeholder='Manufacturer' onChange={handleChange} value={values.manufacturer}/>

              {touched.price && errors.price ? <div style={{color:"red",display:'inline'}}>{errors.price}</div> :<></>}
              <Input type="text" name="price" placeholder='Price' onChange={handleChange} value={values.price}/>

              {touched.quantity && errors.quantity ? <div style={{color:"red",display:'inline'}}>{errors.quantity}</div> :<></>}
              <Input type="text" name="quantity" placeholder='Quantity' onChange={handleChange} value={values.quantity}/>

              <Button type="submit">SUBMIT</Button>
              <Button onClick={handleReset} style={{backgroundColor:"rgb(125,30,30)"}}>Reset</Button>
            </Form>
          </Wrapper>
        </Container>

    </div>
  )
}

export default AddProduct