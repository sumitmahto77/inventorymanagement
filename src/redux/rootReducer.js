import {combineReducers} from "redux";
import {persistReducer} from "redux-persist";
import storage from "redux-persist/lib/storage";
import {authReducer,prodReducer} from "./reducer";




// const rootReducer = combineReducers({
//     users: usersReducers
// });

// export default rootReducer; 



const persistConfig = {
    key : "root",
    storage,
    whitelist: ["auth","prod"]
}


const rootReducer = combineReducers({
    auth : authReducer,
    prod : prodReducer,
});

export default persistReducer(persistConfig, rootReducer); 

