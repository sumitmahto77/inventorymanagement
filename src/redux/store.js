import {createStore, applyMiddleware} from "redux";
import {composeWithDevTools} from 'redux-devtools-extension';
import logger from 'redux-logger';
import reduxThunk from 'redux-thunk';
import {persistStore} from "redux-persist";
import rootReducer from './rootReducer';

const middlewares =[reduxThunk];

if (process.env.NODE_ENV === "development"){
    middlewares.push(logger)
}

export const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(...middlewares)) );

export const persistor = persistStore(store); 


