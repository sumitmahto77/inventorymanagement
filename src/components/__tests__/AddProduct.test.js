import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom'
import {Provider} from 'react-redux';
import {store} from '../../redux/store';
import AddProduct from '../AddProduct';

describe('Add Product component', () =>{
    test('Renders correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <Router>
                <AddProduct />
                </Router>
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
});