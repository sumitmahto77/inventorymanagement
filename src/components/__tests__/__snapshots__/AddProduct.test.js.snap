// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Product component Renders correctly 1`] = `
<div>
  <div
    className="sc-bczRLJ hpVfXr"
  >
    <div
      className="sc-gsnTZi hlGoBi"
    >
      <h1
        className="sc-dkzDqf cBSyNJ"
      >
        Enter product details
      </h1>
      <form
        className="sc-hKMtZM cfGbOi"
        onSubmit={[Function]}
      >
        <input
          className="sc-eCYdqJ liMeyO"
          name="name"
          onChange={[Function]}
          placeholder="name"
          type="text"
          value=""
        />
        <input
          className="sc-eCYdqJ liMeyO"
          name="imgLink"
          onChange={[Function]}
          placeholder="Image Link"
          type="text"
          value=""
        />
        <input
          className="sc-eCYdqJ liMeyO"
          name="description"
          onChange={[Function]}
          placeholder="Description"
          type="text"
          value=""
        />
        <input
          className="sc-eCYdqJ liMeyO"
          name="manufacturer"
          onChange={[Function]}
          placeholder="Manufacturer"
          type="text"
          value=""
        />
        <input
          className="sc-eCYdqJ liMeyO"
          name="price"
          onChange={[Function]}
          placeholder="Price"
          type="text"
          value=""
        />
        <input
          className="sc-eCYdqJ liMeyO"
          name="quantity"
          onChange={[Function]}
          placeholder="Quantity"
          type="text"
          value=""
        />
        <button
          className="sc-jSMfEi jVZLjO"
          type="submit"
        >
          SUBMIT
        </button>
        <button
          className="sc-jSMfEi jVZLjO"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "rgb(125,30,30)",
            }
          }
        >
          Reset
        </button>
      </form>
    </div>
  </div>
</div>
`;
